### DISCUSSION TITLE
Pre-fetching webpack chunks

### DISCUSSION THREAD

**Discussion by marcperrinp** [vercel/next.js#14562] [Help] [answered]
When clicking a specific button on a specific page, I see that some **Webpack chunks** are downloaded. Next time I use the button, the chunks are getting dowloaded again as they already were.

I have little knowledge of how webpack works, but even less about how NextJS uses webpack internally.

My question is: **is a simple way to pre-download those chunks?**

The 3 chunks I would like to pre-fetch:

![image](https://user-images.githubusercontent.com/48444523/85758039-4698cc00-b710-11ea-8a57-3fb33d508693.png)

**Comment by bduffany** [comment]
What does the button do? (Does it change the page route? If so, you could check out [Route prefetching in Next.js](https://web.dev/route-prefetching-in-nextjs/))

**Reply by marcperrinp** [reply]
Hi and thanks :)

No, it runs a piece of script, all on the same page.

**Comment by Timer** [comment]
Are you sure you don't have "Disable cache" checked in your browser's network devtools?

**Reply by marcperrinp** [reply]
Yes, I'm sure :(

**Reply by marcperrinp** [reply]
However, something I should mention is that the piece of code is loaded in `componentDidMount`, maybe that's important to mention? 😛

**Reply by Timer** [reply]
Can you open this as a bug please?

**Reply by marcperrinp** [reply]
@Timer This is not a bug per say: the webpack chunks are getting downloaded and work as expected; however, because I'm using this in the context of a blockchain app, the time it takes to download them derails the request (i.e. timestamp becomes invalid.).

So, after some investigation and to describe my issue in greater details: 

In my frontend, I use a function from an imported node_module - `import xxx from yyyy` -. The first time I use it, my browser downloads some webpack chunks (above screenshot) which is what I would like to avoid (for reason explained above). Hence my question: how would I go about if I wanted to "pre-download" those chunks? I'm open to any other workaround as well!

Thanks a lot for your time 🙏

**Reply by marcperrinp** [reply]
@Timer @bduffany Any idea? The only explanation I can come up with is: when my frontend uses a function imported from a node_module for the first time, it needs to download it. Is this possible? Is there anyway to prevent this and "pre-download" it?

**Comment by Marcpepe** [comment, marked as answer]
Turns out it wasn't coming from my code, but from the api call I was making. Closing this issue.

**Reply by Marcpepe** [reply]
Hmm looks like I can't!
